### PR TITLE
ref #64: Implement scrubbing

### PR DESCRIPTION
**Pull request by akornich**

### CODE REVIEW

File: Rollbar/RollbarClient.cs [hunk 1]
```diff
@@ -23,19 +27,22 @@ public RollbarClient(RollbarConfig config)
             Config = config;
         }
 
-        public RollbarResponse PostAsJson(Payload payload)
+        public RollbarResponse PostAsJson(Payload payload, IEnumerable<string> scrubFileds)
```
Comment (rivkahstandig3636) [inline Rollbar/RollbarClient.cs]: should this be scrubFields? (same comment for the other two times `scrubFileds` is in this file)

File: Rollbar/RollbarQueueController.cs [hunk 2]
```diff
@@ -212,17 +214,25 @@ private void ProcessQueues(AccessTokenQueuesMetadata tokenMetadata)
...
+        {
+            "access_token",
+        };
+
         private RollbarResponse Process(Payload payload, RollbarConfig config)
         {
             var client = new RollbarClient(config);
 
+            IEnumerable<string> safeScrubFields = 
+                JsonScrubber.FilterOutCriticalFields(config.ScrubFields, criticalDataFields);
```
Comment (rokob) [inline Rollbar/RollbarQueueController.cs]: This is not what we want. We only want to keep `access_token` at the top level, i.e. the Rollbar `access_token`. If someone adds `access_token` to the list of scrub fields, we still want to scrub that if it occurs in the payload body.

File: Rollbar/RollbarClient.cs [hunk 3]
```diff
@@ -23,19 +27,22 @@ public RollbarClient(RollbarConfig config)
...
         {
             Assumption.AssertNotNull(payload, nameof(payload));
 
             var jsonData = JsonConvert.SerializeObject(payload);
+            jsonData = JsonScrubber.ScrubJson(jsonData, scrubFileds);
```
Comment (rokob) [inline Rollbar/RollbarClient.cs]: I would have this call a different method than the generic `ScrubJson` like `ScrubJsonPayload` which can pull out the data field, scrub that, and then put the scrubbed data back together with the Rollbar access token.

File: Rollbar/Serialization/Json/JsonScrubber.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            }
+
+            JObject json = JObject.Parse(jsonData);
+
+            ScrubJson(json, scrubFields);
+
+            return json.ToString();
+        }
+
+        private static JToken ScrubJson(JToken json, IEnumerable<string> scrubFields)
```
Comment (rokob) [inline Rollbar/Serialization/Json/JsonScrubber.cs]: Why do these two return something when you never actually use the return value?

File: Rollbar/Serialization/Json/JsonScrubber.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                json.Value = defaultScrubMask;
+                return json;
+            }
+
+            JContainer propertyValue = json.Value as JContainer;
+            if (propertyValue != null)
+            {
+                foreach(var child in propertyValue)
+                {
+                    var result = ScrubJson(child, scrubFields);
```
Comment (rokob) [inline Rollbar/Serialization/Json/JsonScrubber.cs]: Why capture this return value here? It is unused. We should just get rid of the return value altogether.

File: Rollbar/RollbarClient.cs (hunk 1 above)
Comment (rokob) [reply Rollbar/RollbarClient.cs]: Ditto on the typos in the argument names, there are also similar typos in the tests `DataFiledFilteringTest` for example.

Comment (akornich) [reply Rollbar/RollbarClient.cs]: fixed. thanks!

File: Rollbar/Serialization/Json/JsonScrubber.cs (hunk 4 above)
Comment (akornich) [reply Rollbar/Serialization/Json/JsonScrubber.cs]: that was for debugging and possibly unit-testing convenience...

File: Rollbar/RollbarQueueController.cs (hunk 2 above)
Comment (akornich) [reply Rollbar/RollbarQueueController.cs]: done.

File: Rollbar/RollbarClient.cs (hunk 3 above)
Comment (akornich) [reply Rollbar/RollbarClient.cs]: done.

File: Rollbar/Serialization/Json/JsonScrubber.cs (hunk 5 above)
Comment (akornich) [reply Rollbar/Serialization/Json/JsonScrubber.cs]: done

File: Rollbar/Serialization/Json/JsonScrubber.cs (hunk 4 above)
Comment (akornich) [reply Rollbar/Serialization/Json/JsonScrubber.cs]: done

**Pull request merged**